tests/watcher/watcher.go

```diff
@@ -274,3 +272,18 @@ const (
 	// Watch since the resourceVersion passed in to the builder
 	watchSinceResourceVersion startType = "watchSinceResourceVersion"
 )
+
+func shouldIgnoreWatchErr(status *metav1.Status) bool {
+	//api server sometimes closes connections to Watch() client command
+	//ignore this error, because it will reconnect automatically
+	if status.Message == "an error on the server (\"unable to decode an event from the watch stream: http2: response body closed\") has prevented the request from succeeding" {
+		return true
+	}
+	// https://sdk.operatorframework.io/docs/faqs/#i-see-the-warning-in-my-operators-logs-the-resourceversion-for-the-provided-watch-is-too-old-whats-wrong
```

Comment:
I can be wrong but I am pretty certain that the reconnect is done by the informer like this: If connection fails, it lists again and then creates a new watch with the desired resource version as a start. When we swallow this error, based on the usage of the watch, we either miss a warning which we want to let the test fail, or we may miss the actual desired state update and stall the test until it hits a timeout.

Again, I am not 100% sure but almost that we need to re-establish the watch when we try to handle this error.

Reply:
Ref for the case you mention is here https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/request.go#L714 but it does not cover old rv